File: flutter_app/lib/models/activity_model.dart
enum ActivityStatus { ongoing, done }

class Activity {
  int id;
  String name;
  String image;
  String destination;
  double price;
  ActivityStatus status;
  LocationActivity? location;
  Activity({
    required this.name,
    required this.destination,
    required this.image,
    required this.price,
    required this.id,
    this.location,
    this.status = ActivityStatus.ongoing,
  });

  String toString() {
    return 'Activity{id: $id, name: $name, destination: $destination, price: $price}';
  }

  Activity.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        image = json['image'],
        destination = json['destination'],
        price = json['price'].toDouble(),
        location = LocationActivity(
          address: json['address'],
          latitude: json['latitude'],
          longitude: json['longitude'],
        ),
        status =
            json['status'] == 0 ? ActivityStatus.ongoing : ActivityStatus.done;

  Map<String, dynamic> toJson() {
    Map<String, dynamic> value = {
      'id': id,
      'name': name,
      'image': image,
      'destination': destination,
      'price': price,
      'address': location?.address,
      'longitude': location?.longitude,
      'latitude': location?.latitude,
      'status': status == ActivityStatus.ongoing ? 0 : 1
    };
    value['id'] = id;
    return value;
  }
}

class LocationActivity {
  String? address;
  double? longitude;
  double? latitude;
  LocationActivity({
    this.address,
    this.longitude,
    this.latitude,
  });
}

File: go/internal/models/activityModel.go
package models

type Activity struct {
	//gorm.Model
	ID          uint    `gorm:"primary_key; not null" json:"id"`
	Name        string  `gorm:"size:255; not null" json:"name"`
	Image       string  `gorm:"size:255; not null" json:"image"`
	Destination string  `gorm:"size:64; not null" json:"destination"`
	Price       float64 `gorm:"not null" json:"price"`
	Status      int64   `gorm:"default:0;not null" json:"status"`
	Address     string  `gorm:"size:255; not null;" json:"address"`
	Longitude   float64 `gorm:"default:0;not null" json:"longitude"`
	Latitude    float64 `gorm:"default:0;not null" json:"latitude"`
}
